Source/SwiftyDropbox/Shared/Handwritten/DropboxTransportClient.swift

```diff
@@ -529,35 +529,52 @@ public class BaseHosts: NSObject {
     @objc
     let contentHost: String
     @objc
+    let downloadContentHost: String
+    @objc
     let notifyHost: String
 
     @objc
-    public required init(
+    public convenience init(
         apiHost: String,
         contentHost: String,
         notifyHost: String
     ) {
+        self.init(apiHost: apiHost, contentHost: contentHost, downloadContentHost: contentHost, notifyHost: notifyHost)
+    }
+
+    @objc public required init(
+        apiHost: String,
+        contentHost: String,
+        downloadContentHost: String,
+        notifyHost: String
+    ) {
+
         self.apiHost = apiHost
         self.contentHost = contentHost
+        self.downloadContentHost = downloadContentHost
         self.notifyHost = notifyHost
     }
 
     public static var `default`: Self {
         .init(
             apiHost: ApiClientConstants.apiHost,
             contentHost: ApiClientConstants.contentHost,
+            downloadContentHost: ApiClientConstants.contentHost,
             notifyHost: ApiClientConstants.notifyHost
         )
     }
 }
 
 extension BaseHosts {
-    func url(for host: RouteHost) -> String {
+    func url(for attr: RouteAttributes) -> String {
         {
-            switch host {
+            switch attr.host {
             case .api:
                 return apiHost
             case .content:
```

Comment:
this might read a bit better as `case .content where attr.style == .download:`, but YMMV